src/emitter.rs

```diff
@@ -37,7 +39,7 @@ pub(crate) trait Emitter {
         Ok(())
     }
 
-    fn emit_footer(&self, _output: &mut dyn Write) -> Result<(), io::Error> {
+    fn emit_footer(&self, _output: &mut dyn Write, _has_diff: bool) -> Result<(), io::Error> {
```

Comment:
This is the change mentioned in the PR discription. In the case of JSON, I needed an indicator of whether there were any diffs in order to ensure the full emitted result would be valid/parseable JSON (more details inline in `src/emitter/json.rs`)